### PR TITLE
refactered register and login test

### PR DESCRIPTION
**Pull request by nirajacharya2**
Refactored tests according to reviews

### CODE REVIEW

File: tests/e2e/stepDefinitions/registerContext.js [hunk 1]
```diff
@@ -13,34 +17,35 @@ Given("user has navigated to the Setup page", async function () {
...
+  "user sign up with username {string}, password {string} and confirm password {string}",
   async function (username, password, cPassword) {
     await registerPage.submitRegisterData(username, password, cPassword);
   }
 );
 
 Then("user should see message {string}", async function (message) {
-  let a = await registerPage.getErrorMessage(message);
+  let actualErrors = await registerPage.getErrorMessage(message);
+  let featureError = await message.split(",");
```
Comment (PrajwolAmatya) [inline tests/e2e/stepDefinitions/registerContext.js]:
```suggestion
  let actualErrorMessage = await registerPage.getErrorMessage(message);
  let expectedErrorMessage = await message.split(",");
```

File: tests/e2e/stepDefinitions/registerContext.js [hunk 2]
```diff
@@ -13,34 +17,35 @@ Given("user has navigated to the Setup page", async function () {
...
   assert.deepEqual(
-    a[0],
-    a[1],
-    `Expected to get error but got diffrent errors: ${a[0]}`
+    actualErrors,
+    featureError,
+    `Expected to get ${featureError} but got diffrent errors: ${actualErrors}`
```
Comment (PrajwolAmatya) [inline tests/e2e/stepDefinitions/registerContext.js]:
```suggestion
    actualErrorMessage,
    expectedErrorMessage,
    `Expected to get message "${expectedErrorMessage}" but got "${actualErrorMessage}"`
```

File: tests/e2e/stepDefinitions/registerContext.js [hunk 3]
```diff
@@ -13,34 +17,35 @@ Given("user has navigated to the Setup page", async function () {
...
     page.url(),
-    await registerPage.loginURL,
+    await loginPage.loginPageUrl,
     `Expected user to redirect to login page but redirected to page with url ${page.url()}`
   );
 });
 
 Then("user should get success message {string}", async function (succesMsg) {
-  let errors = await registerPage.getSuccessMsg();
+  let errors = await registerPage.getSuccessMsg(succesMsg);
```
Comment (PrajwolAmatya) [inline tests/e2e/stepDefinitions/registerContext.js]: Better using meaningful variable names.
```suggestion
  const actualMessage = await registerPage.getSuccessMsg(succesMsg);
```

**Review by PrajwolAmatya — CHANGES_REQUESTED** [review]
Add description on the PR.

File: tests/e2e/stepDefinitions/registerContext.js [hunk 4]
```diff
@@ -13,34 +17,35 @@ Given("user has navigated to the Setup page", async function () {
...
   async function (username, password, cPassword) {
     await registerPage.submitRegisterData(username, password, cPassword);
   }
 );
 
-Then("user should see message {string}", async function (message) {
-  let a = await registerPage.getErrorMessage(message);
+Then("the user should see message {string}", async function (message) {
+  let actualErrorMessage = await registerPage.getErrorMessage(message);
+  let expectedErrorMessage = await message.split(",");
```
Comment (PrajwolAmatya) [inline tests/e2e/stepDefinitions/registerContext.js]:
```suggestion
  const actualErrorMessage = await registerPage.getErrorMessage(message);
  const expectedErrorMessage = await message.split(",");
```

File: tests/e2e/stepDefinitions/registerContext.js [hunk 5]
```diff
@@ -13,34 +17,35 @@ Given("user has navigated to the Setup page", async function () {
...
-    await registerPage.loginURL,
+    await loginPage.loginPageUrl,
     `Expected user to redirect to login page but redirected to page with url ${page.url()}`
   );
 });
 
-Then("user should get success message {string}", async function (succesMsg) {
-  let errors = await registerPage.getSuccessMsg();
+Then("the user should get success message {string}", async function (succesMsg) {
+  let actualMessage = await registerPage.getSuccessMsg();
```
Comment (PrajwolAmatya) [inline tests/e2e/stepDefinitions/registerContext.js]:
```suggestion
  const actualMessage = await registerPage.getSuccessMsg();
```

File: tests/e2e/features/register.feature [hunk 6]
```diff
@@ -5,12 +5,12 @@ Feature: Register as super user
...
     Background:
-        Given user has navigated to the Setup page
+        Given the user has navigated to the setup page
 
 
-    Scenario Outline: user tries to register with invalid credentials
-        When user registers with username "<username>" ,password "<password>" and confirm password "<confirm password>"
-        Then user should see message "<message>"
+    Scenario Outline: user tries to sign up with invalid credentials
+        When the user sign up with username "<username>", password "<password>" and confirm password "<confirm password>"
```
Comment (grgprarup) [inline tests/e2e/features/register.feature]:
```suggestion
        When the user signs up with the following details
        | username         | <username>         |
        | password         | <password>         |
        | confirm_password | <confirm_password> |
```
We can easily modify the step when more parameters are added for the signup.

**Review by PrajwolAmatya — APPROVED** [review]

**Pull request merged**